docs/user-guide/extending_argocd_with_plugins.md

```diff
@@ -0,0 +1,132 @@
+# Extend argocd with plugins
+
+## Overview
+
+This guide demonstrates how to write plugins for the
+`argocd` CLI tool. Plugins are a mean to extend `argocd` CLI with new sub-commands,
+allowing for new and custom features which are not included in the main distribution
+of `argocd` CLI.
+
+## Prerequisites
+
+You need to have a working `argocd` binary installed locally. You can follow
+the [cli installation documentation](https://argo-cd.readthedocs.io/en/stable/cli_installation/) to install the binary.
+
+## Create `argocd` plugins
+
+A plugin is a standalone executable file, whose name begins with argocd-.
+To install a plugin, move its executable file to anywhere on your `PATH`.
+
+`argocd` allows plugins to add custom commands such as `argocd my-plugin bla bla` by providing a 
+`argocd-my-plugin` binary in the PATH.
+
+## Limitations
+
+It is currently not possible to create plugins that overwrite existing
+`argocd` commands. For example, creating a plugin such as `argocd-version`
+will cause the plugin to never get executed, as the existing `argocd version`
+command will always take precedence over it. Due to this limitation, it is 
+also not possible to use plugins to add new subcommands to existing `argocd` commands.
+For example, adding a subcommand `argocd cluster upgrade` by naming your plugin 
+`argocd-cluster` will lead to the ignorance of plugin.
```

Comment:
"ignorance of plugin" doesn't sound right. I think it would be more clear with:

`by naming your plugin `argocd-cluster` will cause the plugin to be ignored.`